optimum/intel/openvino/modeling_base.py

```diff
@@ -0,0 +1,220 @@
+        else:
+            model_file_names = [file_name]
+            # If not ONNX then OpenVINO IR
+            if not from_onnx:
+                model_file_names.append(file_name.replace(".xml", ".bin"))
+            file_names = []
+            for file_name in model_file_names:
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                )
+                file_names.append(Path(model_cache_path).name)
+            kwargs["model_save_dir"] = Path(model_cache_path).parent
+            bin_file_name = file_names[1] if not from_onnx else None
+            model = cls.load_model(file_names[0], bin_file_name=bin_file_name)
+        return cls(model, config=config, **kwargs)
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        save_dir: Union[str, Path] = default_cache_path,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        **kwargs,
+    ):
+        """
+        Export a vanilla Transformers model into an ONNX model using `transformers.onnx.export_onnx`.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                Directory from which to load
+            save_dir (`str` or `Path`):
+                Directory where the exported ONNX model should be saved, default to
+                `transformers.file_utils.default_cache_path`, which is the cache directory for transformers.
+            use_auth_token (`str` or `bool`):
+                Is needed to load models from a private repository
+            revision (`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization
+        """
+        # Create a local directory to save the model
+        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir.mkdir(parents=True, exist_ok=True)
+        kwargs["model_save_dir"] = save_dir
+
+        # Get the task to load and export the model with the right topology if available else extract it from the hub
+        if cls.export_feature is not None:
+            task = cls.export_feature
+        else:
+            task = HfApi().model_info(model_id, revision=revision).pipeline_tag
+            if task in ["sentiment-analysis", "text-classification", "zero-shot-classification"]:
+                task = "sequence-classification"
+            elif task in ["feature-extraction", "fill-mask"]:
+                task = "default"
+
+        # TODO: support private models
+        preprocessor = get_preprocessor(model_id)
+        # TODO: Add framework ["pt", "tf"]
```

Comment:
I think we should add export to OpenVINO IR here as well. The thing is that OpenVINO can consume its IR and ONNX (not so efficient as IR). No other formats are possible so far. But to make it more consistent it is better to convert everything to IR no matter where it comes from (PT or TF). This is essentially one command for running the Model Optimizer tool: `mo --input_model model.[onnx|pb]`. It is also required `openvino-dev` package to be installed, i.e. it should be an additional dependency.

Reply:
Maybe I mislead but when did the model gets serialized to the OpenVINO IR here? I see the export to ONNX but there should be a final step when we convert it to the IR as it is done in the _save_pretrained() method.

Reply:
It is nice if openvino-dev is not required for inference on PyTorch models. openvino runtime is a very lightweight package that works almost everywhere. The openvino-optimum demo code (https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/optimum/examples/masked_lm_demo.py) works on an M1 for example. When it was tested, Model Optimizer did not work, but Runtime did. Since for PyTorch models we convert the model to ONNX anyway, it would be nice if we can export these models to IR with the OpenVINO API, the same way as is done here: https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/optimum/optimum/intel/openvino/modeling_ov_utils.py#L354

If we do it that way openvino-dev should probably not be a required dependency, but an extra, installable with `pip install optimum-intel[all]` for example.

Reply:
Because onnx models are compatible with openvino runtime, we currently feed the `read_model` method with our onnx model directly. What would be the advantage of converting our onnx model to the openvino IR format as done in the `_save_pretrained` method ? Would it enable some optimization as when exporting the model with Model Optimizer ?